overlord/devicestate/devicestate_remodel_test.go

```diff
@@ -1039,6 +1051,37 @@ func (s *deviceMgrRemodelSuite) TestRemodelReregLocalFails(c *C) {
 	c.Assert(chg, IsNil)
 }
 
+func installSnap(c *C, st *state.State, yaml string, si *snap.SideInfo) *snap.Info {
+	info := snaptest.MakeSnapFileAndDir(c, yaml, nil, si)
+	snapstate.Set(st, info.InstanceName(), &snapstate.SnapState{
+		SnapType: string(info.Type()),
+		Active:   true,
+		Sequence: snapstatetest.NewSequenceFromSnapSideInfos([]*snap.SideInfo{&info.SideInfo}),
+		Current:  info.Revision,
+	})
+	return info
+}
+
+func installEssentialSnaps(c *C, st *state.State, base string) {
+	installSnap(c, st, "name: pc\nversion: 1\ntype: gadget\n", &snap.SideInfo{
+		SnapID:   snaptest.AssertedSnapID("pc"),
+		Revision: snap.R(1),
+		RealName: "pc",
+	})
+
+	installSnap(c, st, "name: pc-kernel\nversion: 1\ntype: kernel\n", &snap.SideInfo{
+		SnapID:   snaptest.AssertedSnapID("pc-kernel"),
+		Revision: snap.R(1),
+		RealName: "pc-kernel",
+	})
+
+	installSnap(c, st, fmt.Sprintf("name: %s\nversion: 1\ntype: base\n", base), &snap.SideInfo{
+		SnapID:   snaptest.AssertedSnapID("core18"),
+		Revision: snap.R(1),
+		RealName: "core18",
```

Comment:
Shouldn't this be `base` instead?

Reply:
Why would the name be base? That is the type, no?